datafusion/src/physical_plan/expressions/array_agg.rs

```diff
@@ -137,6 +129,39 @@ impl Accumulator for ArrayAggAccumulator {
         }
         Ok(())
     }
+}
+
+impl Accumulator for ArrayAggAccumulator {
+    fn update_batch(&mut self, values: &[ArrayRef]) -> Result<()> {
```

Comment:
I think it would be a good idea as a follow on PR to update these implementations to avoid the use of ScalarValue. 

I'll try find some time this week to do it (I need some coding time :) -- lots of reviewing so far)